### PR TITLE
Added the post "Bundling as-fast-as-possible with Fusebox"

### PR DESCRIPTION
**Pull request by theiliad**

### CODE REVIEW

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 1]
```diff
@@ -0,0 +1,306 @@
+---
+layout: post
+title: "Bundling as-fast-as-possible with Fusebox"
+description: "Build and deploy a sample React app, bundled through Fusebox with Hot-Module-Replacement and serve through Express"
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: "Learn how to build and deploy a React app using Fusebox with Hot-Module-Replacement" - what do you think?

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  image: https://avatars1.githubusercontent.com/u/23119087?v=4&s=100
+  bg_color: "#1d79bf"
+tags:
+- bundling
+- fusebox
+- react
+- node
+- express
+---
+`# "What in the world is bundling?"
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: Headings should be H2 

let's also remove the quotes so:

`## What in the world is bundling?`

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+Bundling in a web app is a **recipe** you'd make to **group and compress** a set of files with the same type in order for your users' browsers to download less files.
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: Can you expand on this a little more. Explain it to someone who has never worked with bundling before.

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+Using bundling tools you can also make those files go through certain **pipes** and come out a different file. (e.g. for .scss files you can apply a Sass pipe to convert them into css, and bundle them into 1 final CSS file)
+
+![Webpack](http://i.imgur.com/Keem1SS.jpg)
+
+Bundling is an important step in making **scalable web-apps** these days, and **Webpack** has been very popular for that.
+
+Today I'm going to introduce to you a new tool that bundles **faster than webpack2**, is **much simpler**, and offers **built-in typescript support**:
+![FuseBox](http://fuse-box.org/static/img/logo.png)
+
+# Meet Fusebox
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: Headings should be H2 :)

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+Fusebox is very fast in bundling (& rebundling), thanks to its "aggressive module caching".
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: fast in bundling => fast at bundling

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+});
+
+// Name of the bundle and what file it starts from
+fuse.bundle("app")
+    .instructions(`>index.ts`);
+
+// Fire it up!
+fuse.run();
+```
+Simple huh?
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: Can you quickly explain what this bundle would do for people who are not familiar with bundling at all.

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+# Let's build a simple React app
+![React](http://i.imgur.com/7KtKvNf.gif)
+First things first you need to install some things
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: some things => some dependencies

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+```sh
+npm install --save react react-dom reflect-metadata
+```
+```sh
+npm install --save-dev babel-core babel-plugin-transform-react-jsx babel-preset-es2015 babel-preset-latest chalk fuse-box typescript uglify-js
+```
+# What did I just install?
+
+  - react
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: I would add a brief explanation for each 

- React: React is a front end library...

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+  - babel libraries to be able to transpile react .jsx files into javascript
+  - fuse-box
+  - uglify-js to minify our final bundle
+
+We're going to have to build some files now. Here's an overview of the final folder structure:
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: build some => create some

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+![Folder Structure](http://i.imgur.com/PEki8LO.jpg)
+
+Let's build the Fusebox configurations; make a file named `fuse.js` with the code below:
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: make a file => create a file

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+```
+# We're ready to make our bundle now
+You can now run:
+```sh
+node fuse
+```
+
+to make a bundle out of this react app. The results will go inside a folder named `dist`.
+
+# Let's make a simple server to show our app
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: `## Serving the application`

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+Now we'll make a simple NodeJS (express) server to just serve our `index.html` file that's generated inside `dist`
```
Comment (kukicado) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: make a simple NodeJS (express) => create a simple Express server

File: _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown (hunk 2 above)
Comment (theiliad) [inline _posts/2017-07-20-bundling-as-fast-as-possible-with-fusebox.markdown]: ok, the quotes though are there as if you are asking me that question

**Pull request closed without merging**